Client/Api/Commands/IDeployResourceCommandStep1.cs

```diff
@@ -18,7 +18,7 @@
 
 namespace Zeebe.Client.Api.Commands
 {
-    public interface IDeployResourceCommandStep1
+    public interface IDeployResourceCommandStep1 : ITenantIdCommandStep<IDeployResourceCommandBuilderStep2>
```

Comment:
❌ Sorry, but I think this will break compatibility. I know it is more convenient especially to reuse the interface, but changing the order of interfaces and types will break backward compatibility. Lets not do this.

There is an Interface *Step3, which is left for optional parameters here you can add the method.

Reply:
Sorry, I'm not seeing an interface for the `DeployResourceCommand` for `*Step3`.

The only one I see is the `IDeployResourceCommandBuilderStep2` which has a comment for optional parameters so I assume you're talking about that one and I've updated the PR to remove the new interface and added the new method there.

I am curious on how adding a new optional parameter to the `IDeployResourceCommandStep1` would break compatibility as the changes.  Feels weird specifying the `tenantId` at the end of the builder as opposed to early on.  I'm not pushing for either but this does effect updating all the other commands on where to place the new option.  At the end or at the beginning.

Reply:
Sorry for the late reply I was busy with the normal Zeebe project work.

> The only one I see is the IDeployResourceCommandBuilderStep2 which has a comment for optional parameters
Yeah sorry I meant this one.

> I am curious on how adding a new optional parameter to the IDeployResourceCommandStep1 would break compatibility. Feels weird specifying the tenantId at the end of the builder as opposed to early on. I'm not pushing for either but this does effect updating all the other commands on where to place the new option. At the end or at the beginning.

Imagine you have already code running/deployed like

Reply:
I just checked how it is done in the java client and it was also added at the end, btw similar how you did with an interface but on a different place https://github.com/camunda/zeebe/blob/main/clients/java/src/main/java/io/camunda/zeebe/client/api/command/DeployResourceCommandStep1.java#L100

Reply:
Sounds good.  The code has been updated to remove the interface.  Let me know if you want to move forward with updating the other commands or if you'd prefer to move back to using the interface and apply before the final steps.

Reply:
Yes lets recreate the interface and to it always on final stage. Sorry for the back and forth.

Reply:
Done.